<?xml version="1.0" ?>
<launch>
  <arg name="robot_ip"               default="franka2"/>
  <arg name="load_gripper"           default="false" />
  <arg name="use_interactive_marker" default="true" />
  <arg name="use_gripper_gui"        default="false" />
  <arg name="load_franka_control"    default="false" />

  <!-- Bringup franka_interactive_bringup.laucnh -->
  <group if="$(arg load_franka_control)">
    <include file="$(find franka_interactive_controllers)/launch/franka_interactive_bringup.launch" >
      <arg name="robot_ip" value="$(arg robot_ip)" />
      <arg name="load_gripper" value="$(arg load_gripper)" />
      <arg name="use_gripper_gui" value="$(arg use_gripper_gui)" />
      <arg name="bringup_rviz" value="true" />
    </include>
  </group>
  
  <!-- Loads controller parameters -->
  <rosparam command="load" file="$(find franka_interactive_controllers)/config/franka_interactive_controllers.yaml" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  args="hybrid_joint_impedance_controller"/>
  <param name="hybrid_joint_impedance_controller/pinocchio_urdf_path" value="$(find franka_interactive_controllers)/urdf/panda/panda.urdf"/>

  <!-- IF interactive marker= true: Use interactive marker to define desired pose for impedance (stiffness + damping compensation) controller -->
  <group if="$(arg use_interactive_marker)">
    <node name="interactive_marker" pkg="franka_interactive_controllers" type="interactive_marker.py" required="true" output="screen">
      <param name="link_name" value="panda_link0" />
      <remap from="desired_pose" to="/hybrid_joint_impedance_controller/desired_pose" />
    </node>
  </group>
</launch>